namer/consul/src/main/scala/io/buoyant/namer/consul/DcServices.scala

```diff
@@ -62,7 +62,14 @@ private[consul] object DcServices {
             // may consider retrying certain failures (with backoff).
             state() = Activity.Failed(e)
             stats.errors.incr()
-            Future.exception(e)
+            log.debug("consul connection failed: %s", name)
+
+            e match {
```

Comment:
Instead of adding an extra layer of retrying here, would it be possible to add ConnectionFailedException to the RetryPolicy in BaseApi.infiniteRetryFilter so that it will retry on ConnectionFailedExceptions instead of giving up?